bench/irmin-pack/tree.ml

```diff
@@ -0,0 +1,398 @@
+open Lwt.Infix
+open Bench_common
+
+type key = string list [@@deriving yojson]
+type hash = string [@@deriving yojson]
+type message = string [@@deriving yojson]
+
+type op =
+  | Add of key * string
+  | Remove of key
+  | Find of key * bool
+  | Mem of key * bool
+  | Mem_tree of key * bool
+  | Commit of hash * int64 * message * hash list
+  | Checkout of hash
+  | Copy of key * key
+[@@deriving yojson]
+
+let commits = ref [||]
```

Comment:
This global state feels odd: why not pass it from the `prepare` step to the benchmarks directly? Likewise for the `operation_list`.

Reply:
Yes, thank you.